app/SwiftMailer/SwiftMailerServiceProvider.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\SwiftMailer;
+
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
+use Swift_Events_SimpleEventDispatcher;
+use Swift_Mailer;
+use Swift_StreamFilters_StringReplacementFilterFactory;
+use Swift_Transport_Esmtp_Auth_CramMd5Authenticator;
+use Swift_Transport_Esmtp_Auth_LoginAuthenticator;
+use Swift_Transport_Esmtp_Auth_PlainAuthenticator;
+use Swift_Transport_Esmtp_AuthHandler;
+use Swift_Transport_EsmtpTransport;
+use Swift_Transport_StreamBuffer;
+
+final class SwiftMailerServiceProvider extends AbstractServiceProvider
+{
+    protected function getProvidedServiceNames(): array
+    {
+        return ['mailer'];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared(
+            'mailer',
+            static function () use ($container) {
+                $transport = new Swift_Transport_EsmtpTransport(
+                    new Swift_Transport_StreamBuffer(
+                        new Swift_StreamFilters_StringReplacementFilterFactory()
+                    ),
+                    [
+                        new Swift_Transport_Esmtp_AuthHandler(
+                            [
+                                new Swift_Transport_Esmtp_Auth_CramMd5Authenticator(),
+                                new Swift_Transport_Esmtp_Auth_LoginAuthenticator(),
+                                new Swift_Transport_Esmtp_Auth_PlainAuthenticator(),
+                            ]
+                        ),
+                    ],
+                    new Swift_Events_SimpleEventDispatcher()
+                );
+
+                $options = array_replace(
+                    [
+                        'host' => 'localhost',
+                        'port' => 25,
+                        'username' => '',
+                        'password' => '',
+                        'encryption' => null,
+                        'auth_mode' => null,
+                    ],
+                    $container->get('config')['swiftmailer.options']
+                );
+
+                $transport->setHost($options['host']);
+                $transport->setPort($options['port']);
+                $transport->setEncryption($options['encryption']);
+
+                // The following methods are "magic" methods which PHPStan does not recognize on this class for some
```

Comment:
I think PHPStan is correct because they are only part of the docblock from `Swift_SmtpTransport` and not from `Swift_Transport_EsmtpTransport`. But it is perfect to make PHPStan ignore these.

Reply:
Aha you're right it's a mistake in the docblock(s) of the `swiftmailer/swiftmailer` package. The `Swift_Transport_EsmtpTransport` class defines the magic `_call()` method and supports these methods, but instead the `Swift_SmtpTransport` class that extends it has the docblock (and we don't use that class). But yeah there is no real solution on our end except making PHPStan ignore the errors I think. Or we need to extend `Swift_Transport_EsmtpTransport` with a custom class and define the docblock with magic methods on that one, but that feels very overkill 😄